rpc/client.go

```diff
@@ -32,7 +47,80 @@ func UploadReport(ctx context.Context, opts UploadReportOptions) error {
 	if err := opts.convertAndUploadArtifacts(ctx); err != nil {
 		return errors.Wrap(err, "uploading tests for report")
 	}
-	return errors.Wrap(uploadTests(ctx, gopb.NewCedarPerformanceMetricsClient(opts.ClientConn), opts.Report, opts.Report.Tests, opts.DryRun), "uploading tests for report")
+	if opts.SendToCedar {
+		if err := uploadTests(ctx, gopb.NewCedarPerformanceMetricsClient(opts.CedarClientConn), opts.Report, opts.Report.Tests, opts.DryRun); err != nil {
+			return errors.Wrap(err, "uploading metrics for report")
+		}
+	}
+	if !opts.DryRun && opts.SendRatioSPS > 0 {
+		// This is a random number generator that is used to determine if the report should be uploaded to the new SPS endpoint.
+		// We're using a random number generator that will be set through a config value.
+		randomNumber := rand.Intn(opts.SendRatioSPS)
+		if randomNumber != 0 {
+			return nil
+		}
+		if opts.HTTPClient == nil {
+			opts.HTTPClient = utility.GetHTTPClient()
+			defer utility.PutHTTPClient(opts.HTTPClient)
+		}
+		if err := uploadTestsToSPS(ctx, opts.Report, opts.HTTPClient, opts.SPSURL); err != nil {
+			if opts.SendToCedar {
+				// We'll investigate any errors using Honeycomb. The data is already in Cedar, so we don't need to worry about it.
```

Comment:
If you intend to use honeycomb for logging, is it worth sending to splunk? (The debug level might be better if it's just for back-up logging (assuming this logging would go to the Evergreen index), and I would suggest `grip.WrapError(err, message.Fields` instead of logging the error as a message field).

Reply:
Splunk is likely actually the easier way to go about finding this information. I updated the comment accordingly to call out that we'll be looking at Splunk and implemented the rest of your comment. Let me know if that's the right way to use those grip functions, it seems to log out correctly locally.

Reply:
Debug statements also are but we can filter them at a different level so they won't cause as much clutter. Since this is a part of the codebase your team will own, I'd recommend that you make a Splunk alert on this logging, or else it'll just be logging for no reason.